units/patch_intent.go

```diff
@@ -378,6 +378,17 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 	event.LogPatchStateChangeEvent(patchDoc.Id.Hex(), patchDoc.Status)
 
 	if canFinalize && j.intent.ShouldFinalizePatch() {
```

Comment:
I don't think this is where we want to do this:
There are two cases 1) if a trigger definition has no status set. In that case we will be finalizing the patch here. 2) if the trigger definition is set to wait for a particular status. In that case we will not finalize the patch here. Rather, there will be logic [added later](https://jira.mongodb.org/browse/EVG-13074) to finalize the patch.

To me it makes more sense to add the parameters within [FinalizePatch](https://github.com/evergreen-ci/evergreen/blob/180be7158ba76f52ede1c5ce69efa83e39480da5/model/patch_lifecycle.go#L279). This way they'll be there no matter where it gets finalized from.